modules/core/src/main/scala/ch/epfl/scala/debugadapter/internal/EvaluationProvider.scala

```diff
@@ -42,22 +44,32 @@ private[internal] class EvaluationProvider(
     scalaEvaluators: Map[ClassEntry, ScalaEvaluator],
     mode: DebugConfig.EvaluationMode,
     logger: Logger
-) extends IEvaluationProvider {
+)(implicit context: ExecutionContext)
+    extends IEvaluationProvider {
 
   private var debugContext: IDebugAdapterContext = _
   private val isEvaluating = new AtomicBoolean(false)
 
+  private def fromTryToFuture[T](t: => Try[T]) = {
+    t match {
+      case Success(value) => Future.successful(value)
+      case Failure(e) => Future.failed(e)
+    }
```

Comment:
Here we want to run the evaluation of `t` asynchronously.

```suggestion
    Future(t.get)
```